iree/compiler/Conversion/LinalgToLLVMGPU/ConvertToLLVM.cpp

```diff
@@ -163,83 +158,33 @@ struct HALInterfaceWorkgroupOpsConverter final
-    LowerToLLVMOptions options(m.getContext(), DataLayout(m));
-    options.overrideIndexBitwidth(64);
-    LLVMTypeConverter converter(m.getContext(), options);
-    // Apply in-dialect lowering first. In-dialect lowering will replace ops
-    // which need to be lowered further, which is not supported by a single
-    // conversion pass.
-    // Run Vector -> Vector transformations ahead of conversion to LLVM.
-    {
-      OwningRewritePatternList patterns(&getContext());
-      vector::populateVectorToVectorCanonicalizationPatterns(patterns);
-      vector::populateVectorSlicesLoweringPatterns(patterns);
-      vector::populateVectorContractLoweringPatterns(
-          patterns,
-          vector::VectorTransformsOptions().setVectorTransformsOptions(
-              vector::VectorContractLowering::OuterProduct));
-      mlir::vector::populateVectorTransposeLoweringPatterns(patterns);
-      (void)applyPatternsAndFoldGreedily(m, std::move(patterns));
-    }
-    {
-      OwningRewritePatternList patterns(&getContext());
-      populateGpuRewritePatterns(patterns);
-      (void)applyPatternsAndFoldGreedily(m, std::move(patterns));
-    }
-    {
-      OwningRewritePatternList llvmPatterns(&getContext());
-      llvmPatterns.insert<ConvertFunc, ConvertIREEBindingOp>(m.getContext(),
-                                                             converter);
-      llvmPatterns
-          .insert<HALInterfaceWorkgroupOpsConverter<
-                      IREE::HAL::InterfaceWorkgroupIDOp, NVVM::BlockIdXOp,
-                      NVVM::BlockIdYOp, NVVM::BlockIdZOp>,
-                  HALInterfaceWorkgroupOpsConverter<
-                      IREE::HAL::InterfaceWorkgroupCountOp, NVVM::GridDimXOp,
-                      NVVM::GridDimYOp, NVVM::GridDimZOp>,
-                  HALInterfaceWorkgroupOpsConverter<
-                      IREE::HAL::InterfaceWorkgroupSizeOp, NVVM::BlockDimXOp,
-                      NVVM::BlockDimYOp, NVVM::BlockDimZOp>>(m.getContext());
-      populateStdToLLVMConversionPatterns(converter, llvmPatterns);
-      populateVectorToLLVMConversionPatterns(converter, llvmPatterns);
-      populateGpuToNVVMConversionPatterns(converter, llvmPatterns);
-      LLVMConversionTarget target(getContext());
-      populateStdToLLVMFuncOpConversionPattern(converter, llvmPatterns);
-      configureGpuToNVVMConversionLegality(target);
-      target.addDynamicallyLegalOp<FuncOp>([&](FuncOp funcOp) {
-        if (isEntryPoint(funcOp)) return false;
-        return true;
-      });
-      if (failed(applyPartialConversion(m, target, std::move(llvmPatterns))))
-        signalPassFailure();
-    }
+void populateLLVMConversionPatterns(MLIRContext *context,
+                                    OwningRewritePatternList &patterns,
+                                    LLVMTypeConverter &converter) {
+  patterns.insert<ConvertFunc, ConvertIREEBindingOp>(context, converter);
+  auto codeGenOptions = getLLVMGPUCodegenOptionsFromClOptions();
```

Comment:
we shouldn't get the option from command line here, it should be passed by the caller.

Reply:
makes sense, will modify the function signature appropriately